main.go

```diff
@@ -154,17 +154,24 @@ func dateParamToEpoch(s string, d int64) int32 {
 	switch s {
 	case "now":
 		return int32(timeNow().Unix())
-	case "yesterday", "today", "tomorrow":
+	case "yesterday", "today", "tomorrow", "midnight", "noon", "teatime":
 		t := timeNow()
 		switch s {
 		case "yesterday":
-			t.AddDate(0, 0, -1)
-		case "tomorrow":
-			t.AddDate(0, 0, 1)
+			t = t.AddDate(0, 0, -1)
+		case "tomorrow", "midnight":
+			t = t.AddDate(0, 0, 1)
 		}
 
 		yy, mm, dd := t.Date()
 		midnight := time.Date(yy, mm, dd, 0, 0, 0, 0, defaultTimeZone)
+		switch s {
+		case "noon":
+			midnight = midnight.Add(time.Hour * 12)
```

Comment:
I'm talking specifically about the days when DST changes.  `teatime` is still 4pm, but it's now 15 hours away from midnight.  The other day, when it ticks off there are two 2am hours which means that 4pm is now *17* hours away.